### ISSUE TITLE
Failed at bower install & grunt build

### ISSUE THREAD

**Issue by dev-augmenta** [beeman/loopback-angular-admin#156]
When I try to manually push the repo to heroku I receive this error and the deploy fail.
When I use the button "deploy to heroku" found on github this doesn't happen.
But when I try to push my changes to the project I receive this error.
I tried to add bower cache clean in postinstall but doesn't help...

remote:        npm ERR! Linux 3.13.0-71-generic
remote:        npm ERR! argv "node" "/tmp/build_f6d4937dfe38ee8b0eea2c95b856dd30
/.heroku/node/bin/npm" "install" "--unsafe-perm" "--userconfig" "/tmp/build_f6d4
937dfe38ee8b0eea2c95b856dd30/.npmrc"
remote:        npm ERR! node v0.10.41
remote:        npm ERR! npm  v2.1.18
remote:        npm ERR! code ELIFECYCLE
remote:        npm ERR! loopback-angular-admin@0.1.1 postinstall: `bower install
 && grunt build`
remote:        npm ERR! Exit status 6
remote:        npm ERR!
remote:        npm ERR! Failed at the loopback-angular-admin@0.1.1 postinstall s
cript 'bower install && grunt build'.
remote:        npm ERR! This is most likely a problem with the loopback-angular-
admin package,
remote:        npm ERR! not with npm itself.
remote:        npm ERR! Tell the author that this fails on your system:
remote:        npm ERR!     bower install && grunt build
remote:        npm ERR! You can get their info via:
remote:        npm ERR!     npm owner ls loopback-angular-admin
remote:        npm ERR! There is likely additional logging output above.

**Comment by dev-augmenta** [comment]
so to fix my various publishing issues and complete a push to heroku succesfully I had to complete various step. I'll leave them here just in case someone is facing my same problems.

1) in package.json -> scripts ->  "postinstall": "bower cache clean && bower install && grunt build"
...this will clean bower cache and prevent some errors

2) in package.json update node and npm engines version to yours values. In my case
"engines": {
    "node": "5.4.0",
    "npm": "3.3.12"
  }

3) had support for a different session storage method: in my case
npm install connect-mongostore --save
npm install express-session-mongo --save

4) create a providers.json with values for local
{
  "local": {
    "provider": "local",
    "module": "passport-local",
    "usernameField": "username",
    "passwordField": "password",
    "authPath": "/auth/local",
    "successRedirect": "/auth/account",
    "failureRedirect": "/local"
  }
}

5) in /boot/passport.js add this
var MongoStore = require('express-session-mongo');

around line 22 modify the session storage like this
app.use(loopback.session({
	store : new MongoStore({
		db : 'your_mongo_db_name',
		ip : 'ip_or_url_of_mongodb_server',
		port :  'your_mongodb_port',
		username : 'mongodb_username',
		password: 'mongodb_password'
	}),
    secret: app.get('cookieSecret'),
    saveUninitialized: true,
    resave: true
  }));

6) in /boot/storage.js create a dataSource different from local and storage folder. In my case in heroku that path didn't exists and I was unable to proceed. So I created a storage on Amazon.

7) create or modify the container model in /commons/models

8) in my case I had to install grunt and grunt-cli globally as well
npm install -g grunt grunt-cli

And that's all.
